packages/uploadthing/test/client.test.ts

```diff
@@ -48,20 +48,24 @@ export const setupUTServer = (task: Test) => {
     }),
   );
 
-  const port = genPort(task);
-  const server = app.listen(port);
+  const server = app.listen(0);
+  const port = (server.address() as AddressInfo).port;
+  await new Promise((res) => server.once("listening", res));
 
   const uploadFiles = genUploader<typeof router>({
     package: "vitest",
     url: `http://localhost:${port}`,
   });
 
-  return { uploadFiles, close: () => server.close() };
+  return {
+    uploadFiles,
+    close: () => new Promise((res) => server.close(res)),
+  };
 };
 
 describe("uploadFiles", () => {
-  it("uploads with presigned post", async ({ db, task }) => {
-    const { uploadFiles, close } = setupUTServer(task);
+  it("uploads with presigned post", async ({ db }) => {
+    const { uploadFiles, close } = await setupUTServer();
```

Comment:
Didn't we have a lint rule for no floating promises?

Reply:
it wasn't a promise before. nodejs callback api :)